Project

ovnazarova opened the pull request:
Добавлены 9 файлов:
Archive - класс, хранящий архивы
Note - класс, хранящий заметки
File - интерфейс, который реализуют Archive и Note
MenuState - enum-класс, содержащий константы: типы экранов, между которыми проходит навигация
main - содержится сама система навигации между экранами
ArchiveMenuItem - класс для создания строк меню.
Menu - класс, который содержит функцию, собирающую само меню в зависимости от переданных параметров

ovnazarova committed:
Project

ArturNurtdinov commented on src/main/kotlin/File.kt:
```diff
@@ -0,0 +1,3 @@
+interface File {
+    val name: String
+}
```
Не нужно объединять архив и заметку под одним интерфейсом только из-за одного общего поля - имени. Ты этим не пользуешься да и в целом архив и заметка - интуитивно разные вещи

ArturNurtdinov commented on src/main/kotlin/Main.kt:
```diff
@@ -1,3 +1,71 @@
+import java.util.Scanner
+
+fun main() {
+    val archiveList = mutableListOf<Archive>()
```
Всю логику в функции main я бы ожидал увидеть в классе Menu

ArturNurtdinov commented on src/main/kotlin/Main.kt:
```diff
@@ -1,3 +1,71 @@
+    while (true) {
+        when (menuState) {
+            MenuState.ARCHIVE -> menu.render(archiveList, "Архив")
+            MenuState.NOTES -> menu.render(archiveList[selectedArchiveIndex].notes, "Записку")
```
Почему "Записку", а не "Заметка"?

ArturNurtdinov commented on src/main/kotlin/Main.kt:
```diff
@@ -1,3 +1,71 @@
+            MenuState.ARCHIVE -> menu.render(archiveList, "Архив")
+            MenuState.NOTES -> menu.render(archiveList[selectedArchiveIndex].notes, "Записку")
+            MenuState.VIEW -> {
+                if (selectedNoteIndex >= 0) {
+                    println(archiveList[selectedArchiveIndex].notes[selectedNoteIndex])
+                }
```
`selectedNoteIndex` ты проверила, а `selectedArchiveIndex` - нет. Да и в целом лучше сравнение делать вот такое:
`if (selectedArchiveIndex in archiveList.indices && selectedNoteIndex in archiveList[selectedArchiveIndex].notes.indices)`

ArturNurtdinov reviewed (commented):
Мне понравилась работа с MenuState, но я бы перенёс всё это в класс Menu, а в Main оставил одну строчку - запуск Menu

ovnazarova committed:
Исправила только критические ошибки

ovnazarova commented:
Не успеваю исправить другие замечания. Я их обязательно исправлю. Мне самой очень интересно это сделать, но чуточку попозже, когда не буду опаздывать. Эти два спринта дались очень тяжело.

ovnazarova committed:
1. Замечание "-Креш приложения при вводе букв вместо цифр при выборе пункта меню:" - не нашла. На текст пользовалась try - catch
2. Замечание "-Креш приложения при вводе отрицательного числа при выборе пункта меню:" - нашла. Ошибка была. Исправила.
3. Замечание "И заметку и архив можно создать с пустым именем" - нашла. Не так прочилала задание. Прошу прощения. Исправила. Цикл do-while
4. Замечание "Заметку можно создать без содержания" - нашла. Не так прочитала задание. Прошу прощения. Исправила. Цикл do-while

ovnazarova commented:
1. Замечание "-Креш приложения при вводе букв вместо цифр при выборе пункта меню:" - не нашла. При проверке на текст пользовалась исключением try - catch
2. Замечание "-Креш приложения при вводе отрицательного числа при выборе пункта меню:" - нашла. Ошибка была. Исправила.
3. Замечание "И заметку и архив можно создать с пустым именем" - нашла. Не так прочилала задание. Прошу прощения. Исправила. Цикл do-while
4. Замечание "Заметку можно создать без содержания" - нашла. Не так прочитала задание. Прошу прощения. Исправила. Цикл do-while
